src/services/ProductTransactions.ts

```diff
@@ -55,14 +88,34 @@ export async function createProdTransactionsInDb(apiReq: api.ProdTransactionReq)
     lastmodified: new Date().toISOString(),
   };
 
+  // create database request for attribute entry
   const attributes: db.ProdTransactionAttrDb[] = [];
 
-  if (apiReq.milkQuality) {
-    attributes.push({
-      attrname: 'milkQuality',
-      attrvalue: apiReq.milkQuality,
+  if (apiReq.productType === 'milk') {
+    if (!apiReq.milkQuality) {
+      invalidFields.push('milkQuality');
+    } else {
+      attributes.push({
+        attrname: 'milkQuality',
+        attrvalue: apiReq.milkQuality,
+      });
+    }
+  }
+
+  // throw error for any invalid fields
+  if (invalidFields.length !== 0) {
+    let errorMsg = 'The following fields are invalid or missing';
+    invalidFields.forEach(function (invalidField) {
+      errorMsg += ', ' + invalidField;
     });
+    throw new Error(errorMsg);
   }
 
-  return db.insertProdTransaction(dbInsertReq, attributes);
+  let newUuid: string;
+  try {
+    newUuid = await db.insertProdTransaction(dbInsertReq, attributes);
+  } catch (e) {
+    throw new Error(unhandledErrorMsg);
```

Comment:
should we also be including the original error message from 'e' so that it is easier for us to debug?

Reply:
I'll log it, but it shouldn't be thrown up, because the clients can see it.